src/fast_check/transform_dts.rs

```diff
@@ -0,0 +1,1352 @@
+
+            new_items.push(ModuleItem::ModuleDecl(
+              ModuleDecl::ExportDefaultDecl(value),
+            ))
+          }
+          ModuleDecl::ExportDefaultExpr(export_default_expr) => {
+            let is_overload =
+              if let Expr::Fn(fn_expr) = &*export_default_expr.expr.clone() {
+                fn_expr.function.body.is_none()
+              } else {
+                false
+              };
+            let should_keep = prev_is_overload && !is_overload;
+            prev_is_overload = is_overload;
+            if should_keep {
+              continue;
+            }
+
+            let name = self.gen_unique_name();
+            let name_ident = Ident::new(name.into(), DUMMY_SP);
+            let type_ann = self
+              .expr_to_ts_type(*export_default_expr.expr.clone(), false, true)
+              .map(type_ann);
+
+            if let Some(type_ann) = type_ann {
+              new_items.push(ModuleItem::Stmt(Stmt::Decl(Decl::Var(
+                Box::new(VarDecl {
+                  span: DUMMY_SP,
+                  kind: VarDeclKind::Const,
+                  declare: true,
+                  decls: vec![VarDeclarator {
+                    span: DUMMY_SP,
+                    name: Pat::Ident(BindingIdent {
+                      id: name_ident.clone(),
+                      type_ann: Some(type_ann),
+                    }),
+                    init: None,
+                    definite: false,
+                  }],
+                }),
+              ))));
+
+              new_items.push(ModuleItem::ModuleDecl(
+                ModuleDecl::ExportDefaultExpr(ExportDefaultExpr {
+                  span: export_default_expr.span,
+                  expr: Box::new(Expr::Ident(name_ident)),
+                }),
+              ))
+            } else {
+              new_items.push(ModuleItem::ModuleDecl(
+                ModuleDecl::ExportDefaultExpr(ExportDefaultExpr {
+                  span: export_default_expr.span,
+                  expr: export_default_expr.expr,
+                }),
+              ))
+            }
+          }
+          // Keep all these
+          ModuleDecl::TsImportEquals(_)
+          | ModuleDecl::TsNamespaceExport(_)
+          | ModuleDecl::TsExportAssignment(_)
+          | ModuleDecl::ExportNamed(_)
+          | ModuleDecl::ExportAll(_) => {
+            prev_is_overload = false;
+            new_items.push(ModuleItem::ModuleDecl(module_decl));
+          }
+        },
+        ModuleItem::Stmt(stmt) => {
+          prev_is_overload = false;
+          if let Stmt::Decl(decl) = stmt {
+            match decl {
+              Decl::TsEnum(mut ts_enum) => {
+                ts_enum.declare = true;
+                new_items
+                  .push(ModuleItem::Stmt(Stmt::Decl(Decl::TsEnum(ts_enum))));
+              }
+              Decl::TsInterface(_)
+              | Decl::TsTypeAlias(_)
+              | Decl::Using(_)
+              | Decl::TsModule(_) => {
```

Comment:
Are we handling TypeScript namespaces? Those can contain expressions. Are there any tests for this and for when they're exported from the module?